src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -771,15 +775,19 @@ public int getParameterMode(int param) throws SQLServerException {
                 // if it is not a stored proc, the param can only be input.
                 return parameterModeIn;
             } else {
-                int n = (int) getParameterInfo(param).get("COLUMN_TYPE");
-                switch (n) {
-                    case 1:
-                        return parameterModeIn;
-                    case 2:
-                        return parameterModeOut;
-                    default:
-                        return parameterModeUnknown;
+                Map<String, Object> info = getParameterInfo(param);
+                if (null != info) {
+                    int n = (int) info.get("COLUMN_TYPE");
+                    switch (n) {
+                        case 1:
+                            return parameterModeIn;
+                        case 2:
+                            return parameterModeOut;
+                        default:
+                            return parameterModeUnknown;
```

Comment:
The `deafult` case can be removed.

Reply:
There's no harm in keeping it, plus static analysis tools would always need it

Reply:
Can't the switch just be removed? There's not really enough results to warrant one.
```java
Map<String, Object> info = getParameterInfo(param);
if (null != info) {
    int n = (int) info.get("COLUMN_TYPE");
    if  (n == 1) {
        return parameterModeIn;
    } else if (n == 2) {
        return parameterModeOut;
    } else {
        return parameterModeUnknown;
    }
}

Reply:
Line 912 has similar code (`if` statements are missing `{`, `}` by the way ), without the last redundant `else`.

Reply:
while we're at this, change the if above
            if (null == procMetadata) {

Reply:
I dont see a big deal using switch or if/else. If keeping switch I don't think we should remove default. Default case should always be there for completeness and maintainability - if nothing then just break and do nothing